v-next/hardhat/src/internal/cli/telemetry/telemetry-consent.ts

```diff
@@ -0,0 +1,82 @@
+import path from "node:path";
+
+import { getConfigDir } from "@ignored/hardhat-vnext-core/global-dir";
+import { isCi } from "@ignored/hardhat-vnext-utils/ci";
+import {
+  exists,
+  readJsonFile,
+  writeJsonFile,
+} from "@ignored/hardhat-vnext-utils/fs";
+
+import { confirmationPromptWithTimeout } from "../prompt/prompt.js";
+
+interface TelemetryConsent {
+  consent: boolean;
+}
+
+/**
+ * Get the user's telemetry consent. If already provided, returns the answer.
+ * If not, prompts the user.
+ * Consent is only asked in interactive environments.
+ * @param isHelpCommand Whether the command typed by the user is the help command or not.
+ * @returns True if the user consents to telemetry, false otherwise.
+ */
+export async function getTelemetryConsent(
+  isHelpCommand: boolean,
+): Promise<boolean> {
+  if (canTelemetryBeEnabled(isHelpCommand) === false) {
+    return false;
+  }
+
+  const telemetryConsentFilePath = await getTelemetryConsentFilePath();
+
+  if (await exists(telemetryConsentFilePath)) {
+    // Telemetry consent was already provided, hence return the answer
+    return (await readJsonFile<TelemetryConsent>(telemetryConsentFilePath))
+      .consent;
+  }
+
+  // Telemetry consent not provided yet, ask for it
+  return requestTelemetryConsent();
+}
+
+function canTelemetryBeEnabled(isHelpCommand: boolean): boolean {
+  return (
+    !isHelpCommand &&
```

Comment:
This was here in v2 because telemitry wasn't done using a subprocess and it was slow. We don't need it anymore.

Reply:
Done in [this commit](https://github.com/NomicFoundation/hardhat/pull/5503/commits/6201713b17dc8a944bbc40ffffe8963224412f06)